app/assets/javascripts/application.js

```diff
@@ -16,23 +16,27 @@
 //- require_self
 
 jQuery.ajaxSetup({
-  'beforeSend': function(xhr) {
+  beforeSend: function(xhr) {
     xhr.setRequestHeader("Accept", "text/javascript");
   }
 });
 
 var searchBox = $("#search_form form input[type=search]");
 var searchBoxValue = searchBox.val();
 
-searchBox.keyup(_.debounce(function () {
-  var input = $(this)
-  var newValue = input.val();
-  if (searchBoxValue !== newValue) {
-    input.submit();
-  }
-  searchBoxValue = newValue;
-}, 200));
-
-$( "form" ).on( "submit", function() {
-  history.pushState({}, '', '?' + $(this).serialize());
-});
\ No newline at end of file
+searchBox.keyup(
+  _.debounce(function() {
+    var input = $(this);
+    var newValue = input.val();
+    if (searchBoxValue !== newValue) {
+      input.submit();
+      var state = $(this.form)
+        .serializeArray()
+        .filter(function(e) {
+          return e.name !== "utf8" && e.value !== "";
```

Comment:
suppresses the utf8 & empty values from the resulting string. Seems cleaner, maybe.